rllib/algorithms/cql/torch/cql_torch_learner.py

```diff
@@ -231,23 +290,42 @@ def compute_loss_for_module(
             ],
             dim=1,
         )
-
         cql_loss = (
             torch.logsumexp(q_repeat / config.temperature, dim=1).mean()
             * config.min_q_weight
             * config.temperature
         )
         cql_loss = cql_loss - (q_selected.mean() * config.min_q_weight)
-        # TODO (simon): Implement CQL twin-Q loss here
+
+        critic_loss = sac_critic_loss + cql_loss
+
+        # If a twin Q-value function is implemented calculated its CQL loss.
+        if config.twin_q:
+            q_twin_repeat = torch.cat(
+                [
+                    q_twin_rand_repeat - random_density,
+                    q_twin_next_repeat - logps_next_repeat.detach(),
+                    q_twin_curr_repeat - logps_curr_repeat.detach(),
+                ],
+                dim=1,
+            )
+            cql_twin_loss = (
+                torch.logsumexp(q_twin_repeat / config.temperature, dim=1).mean()
+                * config.min_q_weight
+                * config.temperature
+            )
+            cql_twin_loss - (q_twin_selected.mean()) * config.min_q_weight
```

Comment:
Should this be `-=`?

Could you check all the math also once more? I don't want to miss anything important here :)

Reply:
Oh yes good catch. I hope I just missed the `=` key and its not my eyes :)